creusot-contracts/src/std/iter.rs

```diff
@@ -154,11 +160,14 @@ extern_spec! {
                     where T: IntoIterator<Item = A>, T::IntoIter: Iterator;
             }
 
+            #[pure]
             fn empty<T>() -> Empty<T>;
 
+            #[pure]
```

Comment:
Perhaps in the future we could make a lint which recommends adding `pure` to appropriate functions when you write code (not for extern specs of course).